File: tests/fixtures/positive/android-lint/SecureRandom.java
package com.example;

import java.security.SecureRandom;

class JavaTokenGenerator {
    long literalSeed() {
        SecureRandom rng = new SecureRandom();
        rng.setSeed(1L);
        return rng.nextLong();
    }

    long timeSeed() {
        SecureRandom rng = new SecureRandom();
        rng.setSeed(System.nanoTime());
        return rng.nextLong();
    }
}

File: tests/fixtures/positive/android-lint/SecureRandom.kt
package com.example

import java.security.SecureRandom
import java.util.Random

class TokenGenerator {
    fun generateToken(): Long {
        val rng = Random()
        return rng.nextLong()
    }

    fun generateSeeded(): Long {
        val fq = java.util.Random(42L)
        return fq.nextLong()
    }

    fun deterministicSeed(): Long {
        val rng = SecureRandom()
        rng.setSeed(1234L)
        return rng.nextLong()
    }

    fun timeSeed(): Long {
        val rng = SecureRandom()
        rng.setSeed(System.currentTimeMillis())
        return rng.nextLong()
    }
}
